components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
+ 
+   if ( arguments.length === 1 || arguments.length === 3 ) { 
+       if ( _.isUndefined( states[inportName] ) ) { 
+           return;
+       } 
```

Comment:
Are lines 13-15 needed, given that they return undefined anyway if states[inportName[ is undefined?